generator.rb

```diff
@@ -0,0 +1,93 @@
+# This is the is_vowel? method. It gets used in the run_generator method.
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return true
+  else
+    return false
+  end
+end
+
+# This is the portmanteau generator method.
+# The inputs are collected inside of the method (no parameters passed).
+def run_generator
+  puts "Please enter the first word: "
+  wordA = gets.chomp
+
+  until wordA.length >= 2
+    puts "Oops! This word needs to consist of at least two characters. 
+    Please enter the first word: "
+    wordA = gets.chomp
+  end
+
+  puts "Thanks! Now enter the second word: "
+  wordB = gets.chomp.downcase
+
+  until wordB.length >= 2
+    puts "Oops! This word needs to consist of at least two characters. 
+    Please enter the second word: "
+    wordB = gets.chomp
+  end
+
+  # Instructions:
+  # In the first word, find the last instance of a vowel. The first half of the
+  # portmanteau will keep every letter of the first word until and excluding its last vowel.
+
+  # Approach:
+  # calculate the first partial word by
+  # finding the index of the first vowel
+  # and then generating the partial word using the indices
+  # store the first partial word into variable word_part_A, class of string
+
+  counterA = 0
+
+  wordA.reverse.each_char do |this_letter|
+    if !is_vowel?(this_letter)
+      counterA += 1
+    else
+      break
+    end
+  end
+
+  word_part_A = wordA[0...((wordA.length - 1)) - counterA]
```

Comment:
Hm, not sure if these parens look like what they should look like? Obviously it still works anyway though!